dask/dataframe/io/tests/test_parquet.py

```diff
@@ -2888,3 +2888,108 @@ def test_parquet_pyarrow_write_empty_metadata_append(tmpdir):
+    ddf = ddf.reset_index(drop=True)
+    dd.to_parquet(ddf, tmpdir, engine="pyarrow")
+
+    # Keep the contents of the DataFrame constatn but change the # of partitions
+    ddf2 = ddf.repartition(npartitions=3)
+
+    # Overwrite the existing Dataset with the new dataframe and evaluate
+    # the number of files against the number of dask partitions
+    dd.to_parquet(ddf2, tmpdir, engine="pyarrow", overwrite=True)
+
+    # Assert the # of files written are identical to the number of
+    # Dask DataFrame partitions (we exclude _metadata and _common_metadata)
+    files = os.listdir(tmpdir)
+    files = [f for f in files if f not in ["_common_metadata", "_metadata"]]
+    assert len(files) == ddf2.npartitions
+
+
+def test_read_write_partition_on_overwrite_is_true(tmpdir):
+    # https://github.com/dask/dask/issues/6824
+    from pathlib import Path
+
+    # Create a Dask DataFrame if size (10000, 10) with 5 partitions and write to local, partitioning on the column E
+    df = pd.DataFrame(
+        np.vstack(
+            (
+                np.full((50, 3), 0),
+                np.full((50, 3), 1),
+                np.full((20, 3), 2),
+            )
+        )
+    )
+    df.columns = ["A", "B", "C"]
+    ddf = dd.from_pandas(df, npartitions=5)
+    dd.to_parquet(ddf, tmpdir, engine="pyarrow", partition_on=["A", "B"])
+
+    # Get the total number of files and directories from the original write
+    files_ = Path(tmpdir).rglob("*")
+    files = [f.as_posix() for f in files_]
+    # Keep the contents of the DataFrame constant but change the # of partitions
+    ddf2 = ddf.repartition(npartitions=3)
+
+    # Overwrite the existing Dataset with the new dataframe and evaluate
+    # the number of files against the number of dask partitions
+    # Get the total number of files and directories from the original write
+    dd.to_parquet(
+        ddf2, tmpdir, engine="pyarrow", partition_on=["A", "B"], overwrite=True
+    )
+    files2_ = Path(tmpdir).rglob("*")
+    files2 = [f.as_posix() for f in files2_]
+    # After reducing the # of partitions and overwriting, we expect
+    # there to be fewer total files than were originally written
+    assert len(files2) < len(files)
+
+
+def test_nonempty_path_to_parquet_overwrite(tmpdir):
+    # https://github.com/dask/dask/issues/6824
+    # Create a Dask DataFrame if size (100, 3) with 5 partitions and write to local, partitioning on the column E
+    df = pd.DataFrame(
+        np.vstack(
+            (
+                np.full((50, 3), 0),
+                np.full((50, 3), 1),
+            )
+        )
+    )
+    df.columns = ["A", "B", "C"]
+    ddf = dd.from_pandas(df, npartitions=5)
+    with pytest.raises(ValueError):
+        dd.to_parquet(ddf, "", engine="pyarrow", overwrite=True)
+
+    with pytest.raises(ValueError):
+        dd.to_parquet(ddf, "", engine="pyarrow", overwrite=True)
+
+    with pytest.raises(ValueError):
+        dd.to_parquet(ddf, ".", engine="pyarrow", overwrite=True)
```

Comment:
```suggestion
def test_to_parquet_overwrite_raises(tmpdir, engine):
    # https://github.com/dask/dask/issues/6824
    # Check that overwrite=True will raise an error if the
    # specified path is the current working directory
    df = pd.DataFrame({"a": range(12)})
    ddf = dd.from_pandas(df, npartitions=3)
    with pytest.raises(ValueError):
        dd.to_parquet(ddf, "./", engine=engine, overwrite=True)
    with pytest.raises(ValueError):
        dd.to_parquet(ddf, tmpdir, engine=engine, append=True, overwrite=True)
```

How about something a bit simpler like this?